packages/venia-ui/lib/components/MiniCart/miniCart.gql.js

```diff
@@ -1,17 +1,13 @@
 import gql from 'graphql-tag';
 
-import { ProductListFragment } from './ProductList/productList.gql';
-import { PriceSummaryFragment } from '../CartPage/PriceSummary/priceSummaryFragments';
+import { CartPageFragment } from '../CartPage/cartPageFragments.gql';
 
 export const MiniCartFragment = gql`
     fragment MiniCartFragment on Cart {
         id
-        total_quantity
-        ...ProductListFragment
-        ...PriceSummaryFragment
+        ...CartPageFragment
```

Comment:
You should include `CheckoutPageFragment` for the same reason you need to include `CartPageFragment`. You should also include every top-level fragment explicitly that you know about, even if they overlap:

```suggestion
        ...CartPageFragment
        ...CheckoutPageFragment
        ...ProductListFragment
```

Reply:
Product List Fragment is already part of the cart fragment and sure it is also part of the checkout fragment. I will do this, but I have a bad feeling we are requesting too much stuff in preparations of edge case bugs and that is gonna decrease the performance of the delete operation.

Reply:
> Product List Fragment is already part of the cart fragment and sure it is also part of the checkout fragment. 

This is by coincidence only. The fragment you defined shares properties with the fragment used in cart and checkout to retrieve product details but it is only a coincidence. If cart or checkout decide they need more or less information, the product list in the mini cart may break. 

> I will do this, but I have a bad feeling we are requesting too much stuff in preparations of edge case bugs and that is gonna decrease the performance of the delete operation.

This is the concern I raised in slack yesterday but the decision was made by @jimbo and UX. Perhaps consider adding a comment with some details about why `CartPageFragment` and `CheckoutPageFragment` are included.

Reply:
To be clear, I think it's a problem that the current approach requires us to overfetch unrelated data here in order to ensure that other components reflect new data. I consider that a problem to solve.

However, it's better to have a slow delete operation and accurate data than to have fast operations and inaccurate data. Especially when it comes to price.